public/jobs.js

```diff
@@ -0,0 +1,477 @@
+      editJob.dataset.id = e.target.dataset.id;
+      suspendInput = true;
+      try {
+        const response = await fetch(`/api/v1/jobs/${e.target.dataset.id}`, {
+          method: 'GET',
+          headers: {
+            'Content-Type': 'application/json',
+            Authorization: `Bearer ${token}`,
+          },
+        });
+        const data = await response.json();
+        if (response.status === 200) {
+          company.value = data.job.company;
+          position.value = data.job.position;
+          status.value = data.job.status;
+          showing.style.display = 'none';
+          showing = editJob;
+          showing.style.display = 'block';
+          addingJob.textContent = 'update';
+          message.textContent = '';
+        } else {
+          // might happen if the list has been updated since last display
+          message.textContent = 'The jobs entry was not found';
+          thisEvent = new Event('startDisplay');
+          document.dispatchEvent(thisEvent);
+        }
+      } catch (err) {
+        message.textContent = 'A communications error has occurred.';
+      }
+      suspendInput = false;
+    }
+    //delete option
+    // call the jobs delete API, and in the URL we include the ID of the entry to be deleted.
+    //  Then the home page is displayed again.
+//How do we know it is a delete? In the buildJobsTable function, 
+//each of the delete buttons is given a class of deleteButton. 
+//we check for that class in the e.target.
+// How do we know which entry to delete? The id of the entry is stored
+//  in the data-id of the button, as done in the buildJobsTable function.- lines 35-36
+// How do we do the delete? we need a call to fetch with a method of DELETE giving the URL of that entry.
+//  Be sure we include the authorization header.
+//    fetch is asynchronous, and should be called in a try/catch block.
+// if the delete succeeds  we put a message in the text content of the message paragraph.
+//  Second, we redraw the table showing the updated list of entries.
+//  we can redraw the table by dispatching an event to startDisplay. 
+//if the delete fails we put a message indicating the failure in the message paragraph.
+//don’t want to take input while these asynchronous operations are in progress, 
+//so we set the suspendInput flag before we start them, and clear it afterwards.
+
+
+
+    else if (e.target.classList.contains('deleteButton')) {
+      editJob.dataset.id = e.target.dataset.id;
+      suspendInput = true;
+      try {
+        const response = await fetch(`/api/v1/jobs/${e.target.dataset.id}`, {
+          method: 'DELETE',
+          headers: {
+            'Content-Type': 'application/json',
+            Authorization: `Bearer ${token}`,
+          },
+        });
+        const data = await response.json();
+        if (response.status === 200) {
+          message.textContent = 'This job entry was sucessfully deleted';
```

Comment:
Currently, after I delete a job, it still shows on the page. This would be a good place to kick off the `'startDisplay'` event again, so that the jobs list is refreshed to just show the non-deleted jobs.

Reply:
@akosasante thank you! that was my biggest doubt! I edited this part so now locally it works fine. so I am deploying it to onrender. it takes a while..